Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_on_surface_2.h

```diff
@@ -1295,38 +1294,45 @@ bool remove_vertex
 
 /*! \ingroup PkgArrangementOnSurface2Funcs
  *
- * Compute the zone of the given \f$ x\f$-monotone curve in the existing
- * arrangement. Meaning, it output the arrangement's vertices, edges and faces
- * that the \f$ x\f$-monotone curve intersects. The order of the objects is the
- * order that they are discovered when traversing the \f$ x\f$-monotone curve
- * from left to right.
+ * computes the zone of the given \f$x\f$-monotone curve in a given
+ * arrangement. More precisely, this function finds the arrangement vertices,
+ * edges ,and faces that the given \f$x\f$-monotone curve intersects, and
+ * inserts them in the order they are discovered when traversing the
+ * \f$x\f$-monotone curve from left to right into an output contaiuner given
+ * through an output iterator. An object in the resulting zone is represented by
+ * a discriminated union container that holds a vertex handle, halfedge handle,
+ * or a face handle.
  *
  * A given point-location object is used for answering point-location queries
  * during the insertion process. By default, the function uses the "walk along
- * line" point-location strategy - namely an instance of the class
+ * line" point-location strategy, namely, an instance of the class
  * `Arr_walk_along_line_point_location<Arrangement_on_surface_2<GeometryTraits,
- * TopologyTraits> >`.
+ * TopologyTraits>>`.
  *
- * Compute the zone of the given \f$ x\f$-monotone curve `c` in the arrangement
- * `arr`.
+ * \param arr The given arrangement.
+ * \param c The \f$x\f$-monotone curve.
+ * \param oi The output iterator that points at the output container.
+ * \param pl The point-location object.
+ * \return The past-the-end iterator of the output container.
  *
  * \pre If provided, `pl` must be attached to the given arrangement `arr`.
+ * \pre The instantiated `GeometryTraits` class must model the
+ *      `ArrangementXMonotoneTraits_2` concept.
+ * \pre The point-location object `pl`, must model the
+ *      `ArrangementPointLocation_2` concept.
+ * \pre Dereferencing `oi` must yield a polymorphic object of type
+ *      `std::variant<Arrangement_on_surface_2::Vertex_handle,
+ *                    Arrangement_on_surface_2::Halfedge_handle,
+ *                    Arrangement_on_surface_2::Face_handle>`.
```

Comment:
In the nightly documentation test CGAL-6.0-Ic-67 (https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-6.0-Ic-67/index.html) we get for the 1.9.6 version of doxygen the warnings:
```
/home/cgal-testsuite/cgal_doc_build/CGAL-6.0-Ic-67/doc/Arrangement_on_surface_2/CGAL/Arr_vertical_decomposition_2.h:45: warning: explicit link request to 'variant' could not be resolved
/home/cgal-testsuite/cgal_doc_build/CGAL-6.0-Ic-67/doc/Arrangement_on_surface_2/CGAL/Arrangement_on_surface_2.h:1324: warning: explicit link request to 'variant' could not be resolved
```
(as of doxygen version 1.9.7 this problem has been solved).

As a workaround the problem can be solved to reformulate the:
```
 *      `std::variant<Arrangement_on_surface_2::Vertex_handle,
 *                    Arrangement_on_surface_2::Halfedge_handle,
 *                    Arrangement_on_surface_2::Face_handle>`.
 ```
to
```
 *      `std::variant<Arrangement_on_surface_2::Vertex_handle, Arrangement_on_surface_2::Halfedge_handle, Arrangement_on_surface_2::Face_handle>`.
 ```

analogous for doc/Arrangement_on_surface_2/CGAL/Arr_vertical_decomposition_2.h